### PR TITLE
Pat 401 url refining

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: server/routes/createNeeds.js [hunk 1]
```diff
@@ -46,13 +46,13 @@ module.exports = function PathfinderNeedsRoutes({ offenderService, pathfinderSer
...
       ? getPathFor({ data: payloadFields, config: sectionConfig })(nominalId)
-      : `/nominal/cim-matrix/${nominalId}`
+      : `/nominal/${nominalId}/cim-matrix`
     return location
   }
 
   // Get the next path based on a value or a discriminator method
   const getNextPathFromFunction = pathFunction => {
-    const nextPath = pathFunction('').split('/')[3]
+    const nextPath = pathFunction('').split('/')[4]
```
Comment (lbennett-moj) [inline server/routes/createNeeds.js]: now has extra section on path /nominal/

File: server/routes/standardRouter.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ const express = require('express')
 
 module.exports = ({ authenticationMiddleware }) => {
   return routes => {
-    const router = express.Router()
+    const router = express.Router({ mergeParams: true })
```
Comment (lbennett-moj) [inline server/routes/standardRouter.js]: required to propagate :nominalId to routers when matched in app.use

File: server/app.js [hunk 3]
```diff
@@ -266,15 +266,21 @@ module.exports = function createApp({
 
   app.use('/', standardRoute(createCaseListRouter({ pathfinderService, userService, offenderService })))
   app.use('/api/', standardRoute(createApiRouter({ offenderService })))
+  app.use(
+    `/nominal/:nominalId/documents`,
```
Comment (lbennett-moj) [inline server/app.js]: /document changed to plural to be consistent with other areas

File: server/routes/testutils/appSetup.js [hunk 4]
```diff
@@ -9,7 +9,10 @@ const db = require('../../../server/data/dataAccess/db')
 const nunjucksSetup = require('../../utils/nunjucksSetup')
 const pdfRenderer = require('@ministryofjustice/express-template-to-pdf')
 
-module.exports = route => {
+/* givenPath is an optional alternative path which can be useful for testing routers which
+ have a param that is consumed by the app.use statement - eg /nominal/:nomimalId/review/bandingReview
+ */
+module.exports = (route, givenPath) => {
```
Comment (timharrison-moj) [inline server/routes/testutils/appSetup.js]: Do we use the optional givenPath in tests, or is it for potential future use? I wonder whether its worth expanding the comment to say hows its useful?

Comment (lbennett-moj) [reply server/routes/testutils/appSetup.js]: Its just in tests - this is the testUtil's appSetup.

**Pull request merged**